### DISCUSSION TITLE
Use HikariDataSource for connecting to sqlg

### DISCUSSION THREAD

**Discussion by 29nidhishah** [pietermartin/sqlg#517] [Q&A]
I am using slqlg 2.1.6. I want my customized HikariDataSoure to be used to connect to sqlG and create a graph but I see SqlgHikariDataSource creates it own HikariDataSource. I dont want to modify the sqlg repository code. Is there a method to achieve this?

**Comment by pietermartin** [comment]
No, there is no support to pass in an existing data source. I can add it but I'll add it to the current master branch not `2.1.6`.
Any reason you can not upgrade to `3.1.0`?

**Reply by 29nidhishah** [reply]
I am using java 11 hence cannot upgrade to 3.1.0. Can I use autowire/bean in my class to modify and add my custom HikariDataSource

**Reply by pietermartin** [reply]
Don't know what `autowire/bean` bean is?
But I don't see how it will be possible. Sqlg does have support to look up the data source in JNDI. Maybe that is a way you can try.
